Makefile

```
@@ -0,0 +1,57 @@
+.PHONY: $(MAKECMDGOALS)
+
+clean:
+	@rm -rf tests/tmp
+
+cleantests:
+	@rm -rf tests/tmp/.git
+	@rm -rf tests/tmp/empty
+	@rm -rf tests/tmp/CHANGELOG.md
+
+docs:
+	@mkdocs serve
+
+docs-deploy:
+	@mkdocs gh-deploy
+
+format:
+	@ruff format
+
+gen generate:
+	@bash -c 'source tests/helpers.sh && generate ${PWD} tests/tmp'
+
+reset-history: gen
+	@./reset-history.sh
+
+test: cleantests
+	@./runtests.sh
+
+changelog:
+	@git-changelog -Tio CHANGELOG.md -Bauto -c angular
+
+release:
+	@test -n "$(version)" || { echo "error: usage: make release version=x.y.z" >&2; exit 1; }
+	@git add CHANGELOG.md
+	@git commit -m "docs: Update changelog for version $(version)"
+	@git tag $(version)
```

```suggestion
	@git tag "$(version)"
```
Also, what if the tag already exists? Maybe `-f` for that case?
```suggestion
	@git tag -f "$(version)"
```

---

I didn't go with `-f` yet, as I'd consider it an error if the tag already exists, I don't want to silently overwrite it (think typo when typing the command, it should absolutely not overwrite an existing tag!)